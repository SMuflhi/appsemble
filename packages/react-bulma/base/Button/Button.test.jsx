import { shallow } from 'enzyme';
import React from 'react';

import { Button } from '..';

describe('Button', () => {
  it('should render a button', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });

  it('should be possible to pass custom class names', () => {
    const component = shallow(<Button className="custom" />);
    expect(component).toMatchSnapshot();
  });

  it('should be possible to specify the button as active using a boolean prop', () => {
    const component = shallow(<Button active />);
    expect(component).toMatchSnapshot();
  });

  it('should be possible to specify the button is loading using a boolean prop', () => {
    const component = shallow(<Button loading />);
    expect(component).toMatchSnapshot();
  });

  it('should be possible to specify a color', () => {
    const component = shallow(<Button color="primary" />);
    expect(component).toMatchSnapshot();
  });

  it('should be possible to specify a custom component', () => {
    const component = shallow(<Button component="a" />);
    expect(component).toMatchSnapshot();
  });

  it('should be possible to override the button type', () => {
    const component = shallow(<Button type="submit" />);
    expect(component).toMatchSnapshot();
  });
});
