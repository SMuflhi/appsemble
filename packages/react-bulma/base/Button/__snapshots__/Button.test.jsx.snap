// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be possible to override the button type 1`] = `
<button
  className="button"
  type="submit"
/>
`;

exports[`Button should be possible to pass custom class names 1`] = `
<button
  className="button custom"
  type="button"
/>
`;

exports[`Button should be possible to specify a color 1`] = `
<button
  className="button is-primary"
  type="button"
/>
`;

exports[`Button should be possible to specify a custom component 1`] = `
<a
  className="button"
  type="button"
/>
`;

exports[`Button should be possible to specify the button as active using a boolean prop 1`] = `
<button
  className="button is-active"
  type="button"
/>
`;

exports[`Button should be possible to specify the button is loading using a boolean prop 1`] = `
<button
  className="button is-loading"
  type="button"
/>
`;

exports[`Button should render a button 1`] = `
<button
  className="button"
  type="button"
/>
`;
